nullaway/src/test/java/com/uber/nullaway/jspecify/ArrayTests.java

```diff
@@ -278,6 +278,51 @@ public void arraysAndGenerics() {
         .doTest();
   }
 
+  @Test
+  public void genericArraysReturnedAndPassed() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "  static class Foo<T> {}",
+            "  static class Bar<T> {",
+            "    Foo<T>[] getFoosPositive() {",
+            "      @Nullable Foo<T>[] result = new Foo[0];",
+            "      // BUG: Diagnostic contains: Cannot return expression of type @Nullable Foo<T>[] from method",
+            "      return result;",
+            "    }",
+            "    Foo<T>[] getFoosNegative() {",
+            "      Foo<T>[] result = new Foo[0];",
+            "      return result;",
+            "    }",
+            "    void takeFoos(Foo<T>[] foos) {}",
+            "    void callTakeFoosPositive(@Nullable Foo<T>[] p) {",
+            "      // BUG: Diagnostic contains: Cannot pass parameter of type @Nullable Foo<T>[]",
+            "      takeFoos(p);",
+            "    }",
+            "    void callTakeFoosNegative(Foo<T>[] p) {",
+            "      takeFoos(p);",
+            "    }",
+            "    void takeFoosVarargs(Foo<T>[]... foos) {}",
```

Comment:
To be clear, the type of `foos` here ends up being `Foo<T>[][]` right? With a single element array passed in 311 that gets converted to the type `@Nullable Foo<T>[][]` (meaning a two dimensional array of nullable `Foo`s), correct?

Reply:
Yes, that's right; added a comment in 4aa9da7435062a0e308009ba76a8fa61b47a2a01.  Note that we don't have support for multidim arrays yet in JSpecify mode, so we can't yet test passing in a `@Nullable Foo<T>[][]` array directly.